src/interface/variable.cpp

```diff
@@ -85,28 +86,42 @@ CellVariable<T>::AllocateCopy(const bool allocComms, std::weak_ptr<MeshBlock> wp
 /// Initialize a 6D variable
 template <typename T>
 void CellVariable<T>::allocateComms(std::weak_ptr<MeshBlock> wpmb) {
-  // set up fluxes
   std::string base_name = label();
+
+  // TODO(JMM): Note that this approach assumes LayoutRight. Otherwise
+  // the stride will mess up the types.
+
+  // Compute size of unified flux_data object and create it. A unified
+  // flux_data_ object reduces the number of memory allocations per
+  // variable per meshblock from 5 to 3.
+  int n_outer = 0;
   if (IsSet(Metadata::Independent)) {
-    flux[X1DIR] = ParArrayND<T>(base_name + ".fluxX1", GetDim(6), GetDim(5), GetDim(4),
-                                GetDim(3), GetDim(2), GetDim(1));
-    if (GetDim(2) > 1)
-      flux[X2DIR] = ParArrayND<T>(base_name + ".fluxX2", GetDim(6), GetDim(5), GetDim(4),
-                                  GetDim(3), GetDim(2), GetDim(1));
-    if (GetDim(3) > 1)
-      flux[X3DIR] = ParArrayND<T>(base_name + ".fluxX3", GetDim(6), GetDim(5), GetDim(4),
-                                  GetDim(3), GetDim(2), GetDim(1));
+    n_outer += (GetDim(1) > 1) + (GetDim(2) > 1) + (GetDim(3) > 1);
   }
+  flux_data_ = ParArray7D<T>(base_name + ".flux_data", n_outer, GetDim(6), GetDim(5),
+                             GetDim(4), GetDim(3), GetDim(2), GetDim(1));
 
-  if (wpmb.expired()) return;
+  // set up fluxes
+  int offset = 0;
+  if (IsSet(Metadata::Independent)) {
+    for (int d = X1DIR; d <= X3DIR; ++d) {
+      if (GetDim(d) > 1) {
+        flux[d] = ParArrayND<T>(
+            Kokkos::subview(flux_data_, offset++, Kokkos::ALL(), Kokkos::ALL(),
+                            Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL()));
+      }
+    }
+  }
```

Comment:
Probably a question of style, but I think the original, explicit version above is easier to parse.

```suggestion
  // set up fluxes
  if (IsSet(Metadata::Independent)) {
    flux[X1DIR] = ParArrayND<T>(Kokkos::subview(flux_data_, 0, Kokkos::ALL(), Kokkos::ALL(),
                                Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL()));
    if (GetDim(2) > 1)
      flux[X2DIR] = ParArrayND<T>(Kokkos::subview(flux_data_, 1, Kokkos::ALL(), Kokkos::ALL(),
                                Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL()));
    if (GetDim(3) > 1)
      flux[X3DIR] =ParArrayND<T>(Kokkos::subview(flux_data_, 2, Kokkos::ALL(), Kokkos::ALL(),
                                Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL()));
  }
```

Reply:
I disagree, but I don't feel strongly about it. I can change it back.

Reply:
We could reduce the logic from an `if for if` to a single `for` reusing the information from the previous condition here, see above.

Reply:
Resolved as discussed above